src/fairseq2/datasets/speech.py

```diff
@@ -6,24 +6,172 @@
 
 from __future__ import annotations
 
+import random
 from abc import ABC, abstractmethod
 from dataclasses import dataclass
+from functools import partial, reduce
 from pathlib import Path
-from typing import Final, final
+from typing import Any, Callable, Dict, Final, List
 
+import numpy as np
 import torch
+from torch import Tensor
+from torch.nn.functional import layer_norm
 from typing_extensions import override
 
+from fairseq2.data import Collater, DataPipelineBuilder, FileMapper, create_bucket_sizes
+from fairseq2.data.audio import AudioDecoder, WaveformToFbankConverter
+from fairseq2.data.text import StrSplitter, read_text
 from fairseq2.datasets import (
     DataPipelineReader,
     DataReader,
+    DataReadError,
     DataReadOptions,
     DatasetHubAccessor,
+    DatasetLoadError,
+    LengthBatching,
+    StaticBatching,
+    UnknownSplitError,
 )
 from fairseq2.error import NotSupportedError
 from fairseq2.gang import Gang
+from fairseq2.logging import log
 from fairseq2.models.sequence import SequenceBatch
-from fairseq2.typing import DataType
+from fairseq2.nn.padding import get_seqs_and_padding_mask
+from fairseq2.typing import DataType, Device
+
+try:
+    import torchaudio  # type: ignore
+except ImportError:
+    torchaudio = None
+    log.warning(
+        "torchaudio is not installed. Please install it with `pip install torchaudio`."
+    )
+
+
+# SpecAugment
+@torch.no_grad()
```

Comment:
with some extra comments, it should be ok to go on main

Reply:
Already fully migrated [here](https://github.com/facebookresearch/fairseq2/blob/wav2vec2_trainer/recipes/wav2vec2/ssl/data/dataset.py).